src/threshold_decrypt.rs

```diff
@@ -182,7 +187,7 @@ impl<N: NodeIdT> ThresholdDecrypt<N> {
         let faulty_senders: Vec<N> = self
             .shares
             .iter()
-            .filter(|(id, share)| !self.is_share_valid(id, share))
+            .filter(|(id, (_, share))| !self.is_share_valid(id, share))
```

Comment:
This would pass the same pair of ID and share with different node indices. Would that still be correct? Shouldn't we extend `is_share_valid` to handle node indices?

Reply:
Would it? Maybe I missed something, but my intention was to just compute the index that matches the ID for each entry at the point where we put them into the map instead of here, to avoid the `unwrap`.
For each entry, the `idx` should be exactly the result of `netinfo.node_index(id)`, and `share` should be the one we received from `id`.

Reply:
For example, if node 0 is removed, all other nodes' indices decrement by 1. If it's impossible to get the same share both with the old (before node 0 removal) and new (after removal) node indices, please ignore my remark.

Reply:
I'd say `ThresholdDecrypt` (and in general anything other than DHB and QHB) doesn't support changing sets of validators. The `NetworkInfo` is constant, and the node indices stay the same.